cylc/flow/scripts/workflow_state.py

```diff
@@ -231,15 +227,11 @@ def main(parser: COP, options: 'Values', workflow_id: str) -> None:
             options.status not in CylcWorkflowDBChecker.STATE_ALIASES):
         raise InputError(f"invalid status '{options.status}'")
 
-    # this only runs locally
-    if options.run_dir:
-        run_dir = expand_path(options.run_dir)
-    else:
-        run_dir = get_cylc_run_dir()
+    workflow_id = infer_latest_run_from_id(workflow_id, options.alt_run_dir)
 
     pollargs = {
         'workflow_id': workflow_id,
-        'run_dir': run_dir,
+        'run_dir': expand_path(options.alt_run_dir or get_cylc_run_dir()),
```

Comment:
Doesn't `get_cylc_run_dir` already employ `expand_path`?
This is why I just included an optional arg to `get_cylc_run_dir`, and removed the import of `expand_path` in a couple of places.

Reply:
Yeah, all it does is expand the default run-dir.  I'm considering other options. The various workflow run path functions are getting a bit convoluted...